### PR TITLE
[Merged by Bors] - Export slashing protection per validator 

### PR DESCRIPTION
**Pull request by michaelsproul**
## Issue Addressed

Part of https://github.com/sigp/lighthouse/issues/2557

## Proposed Changes

Refactor the slashing protection export so that it can export data for a subset of validators.

This is the last remaining building block required for supporting the standard validator API (which I'll start to build atop this branch)

## Additional Info

Built on and requires #2598

### CODE REVIEW

**Comment by michaelsproul** [comment]
Needs some tests and a CLI update before merging

**Comment by michaelsproul** [comment]
> Needs some tests and a CLI update before merging

Done. Now just blocked on #2598

**Comment by paulhauner** [comment]
I'll give this a review once #2598 is merged :)

**Comment by michaelsproul** [comment]
Rebased on `unstable` and ready for review once CI passes

**Comment by michaelsproul** [comment]
Re-running CI as `cargo-deps` failed with a network error

**Review by paulhauner — APPROVED** [review]
Beautiful!

**Pull request closed without merging**
